homeassistant/components/sensor/nordpool.py

```diff
@@ -0,0 +1,252 @@
+        self._currency = currency
+        self._currency_fraction = _CURRENCY_FRACTION[currency]
+
+        # Setup region name for correct prices.
+        self._region = region
+
+        # Fetch today's prices.
+        self.fetch_new_data(_TODAY)
+        # Initiate the first value
+        now = datetime.datetime.now() + datetime.timedelta(hours=self._offset)
+        self._state = self._prices[now.hour]
+
+    @property
+    def available(self):
+        """Return the availability of the sensor."""
+        return self._available
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the current power price from Nordpool if available."""
+        return self._state
+
+    @property
+    def unit_of_measurement(self):
+        """Return the correct currency."""
+        return self._currency_fraction
+
+    @property
+    def device_state_attributes(self):
+        """Return aditional price information."""
+        return {
+            'Min': self._min,
+            'Max': self._max,
+            'Average': self._average,
+            'Peak': self._peak,
+            'Off-peak1': self._off_peak1,
+            'Off-peak2': self._off_peak2,
+            'next update': str(self._next[0]) + ':' + str(self._next[1]),
+            ATTR_ATTRIBUTION: "For details, see https://www.nordpoolgroup.com/"
+        }
+
+    def update(self):
+        """Update the price to the current price."""
+        now = datetime.datetime.now() + datetime.timedelta(hours=self._offset)
+        # If the current day is a new day, update the data.
+        if self._day[_TODAY] != int(now.day) and \
+                self._day[_TOMORROW] == int(now.day):
+            self._prices = list(self._tomorrow)
+            self._day[_TODAY] = self._day[_TOMORROW]
+            self._day[_TOMORROW] = 99
+
+        if self._day[_TODAY] != int(now.day):
+            # Fetch today's prices if the current data is invalid.
+            self.fetch_new_data(_TODAY)
+
+        # Fetch new data if it is time to do so.
+        tomorrow = now + datetime.timedelta(days=1)
+        if ((now.hour > self._next[0] or (now.hour >= self._next[0] and
+                                          now.minute >= self._next[1])) and
+                self._day[_TOMORROW] != tomorrow.day):
+            self.fetch_new_data(_TOMORROW)
+
+        # Update the current price if the price-table is valid
+        self._available = False
+        if self._prices[now.hour] is not None:
+            self._available = True
+            self._state = self._prices[now.hour]
+            self._min = self._prices[24]
+            self._max = self._prices[25]
+            self._average = self._prices[26]
+            self._peak = self._prices[27]
+            self._off_peak1 = self._prices[28]
+            self._off_peak2 = self._prices[29]
+
+    def fetch_new_data(self, selected_day):
```

Comment:
All of this request logic should be coming in from a separate pypi library. See [documentation](https://developers.home-assistant.io/docs/en/creating_component_code_review.html#4-communication-with-devices-services)

Reply:
Thanks, I'll implement the nordpool library from pypi.